catboost/libs/fstr/shap_values.h

```diff
@@ -59,27 +74,30 @@ void CalcShapValuesForDocumentMulti(
 
 TShapPreparedTrees PrepareTrees(const TFullModel& model, NPar::TLocalExecutor* localExecutor);
 TShapPreparedTrees PrepareTrees(
-        const TFullModel& model,
-        const NCB::TDataProvider* dataset, // can be nullptr if model has LeafWeights
-        int logPeriod,
-        NPar::TLocalExecutor* localExecutor,
-        bool calcInternalValues = false
+    const TFullModel& model,
+    const NCB::TDataProvider* dataset, // can be nullptr if model has LeafWeights
+    int logPeriod,
+    NPar::TLocalExecutor* localExecutor,
+    bool calcInternalValues = false,
+    EPreCalcShapValues mode = EPreCalcShapValues::Auto
```

Comment:
Let's remove this default and move this parameter after `logPeriod` here and below.